### PR TITLE
chore(build): bump required pipenv

### PR DESCRIPTION
**Pull request by sfoster1**
It would be really nice to have a modern pipenv, so let's see if it works.

## Testing
- See if you can still `make setup-py` with this version of pipenv
- See if you can change something in a `Pipfile` and then `pipenv update`

### CODE REVIEW

**Review by amitlissack — APPROVED** [review]

File: jest.config.js [hunk 1]
```diff
@@ -16,6 +16,7 @@ module.exports = {
     '\\.(jpg|png|gif|svg|woff|woff2|webm)$':
       '@opentrons/components/src/__mocks__/file.js',
   },
+  modulePathIgnorePatterns: ['/shared-data/python/.*'],
```
Comment (mcous) [inline jest.config.js:19]: I'm still getting Jest module collisions. I think ignore list needs to include all our other Python directories as well:

```shell
∙ make test-js
jest \
	--coverage=true \
	--watch=false \
	--updateSnapshot=false \
	--ci=false
jest-haste-map: Haste module naming collision: @opentrons/update-server
  The following files share their name; please adjust your hasteImpl:
    * <rootDir>/update-server/otupdate/package.json
    * <rootDir>/update-server/build/lib/otupdate/package.json

jest-haste-map: Haste module naming collision: @opentrons/api-server
  The following files share their name; please adjust your hasteImpl:
    * <rootDir>/api/build/lib/opentrons/package.json
    * <rootDir>/api/src/opentrons/package.json
```

**Review by mcous — CHANGES_REQUESTED** [review]
Other than additional Jest module collisions on the JS side, this looks to be working for me on the Python side

File: jest.config.js (hunk 1 above)
Comment (sfoster1) [reply jest.config.js:19]: Will do

**Review by mcous — APPROVED** [review]

**Review by ahiuchingau — APPROVED** [review]
`make setup` and `make test-py` still worked after switching to the new pipenv

**Comment by b-cooper** [comment]
worked as expected, tests still pass for py dirs ✅

**Pull request merged**